src/base/global.h

```diff
@@ -48,3 +48,27 @@ constexpr typename std::add_const<T>::type asConst(T &&t) noexcept { return std:
 // Prevent const rvalue arguments
 template <typename T>
 void asConst(const T &&) = delete;
+
+// Replacement for the deprecated`QSet<T> QSet::fromList(const QList<T> &list)`
+// TODO: remove this when minimum requirement is raised to Qt 5.14
+template <typename T>
+QSet<T> toQSet(const QList<T> &list)
+{
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+    return {list.cbegin(), list.cend()};
+#else
+    return QSet<T>::fromList(list);
```

Comment:
Aren't they available in Qt 5.14?

Reply:
Yes, but Qt 5.14 marked them deprecated and they suggest to use `QSet::QSet(InputIterator first, InputIterator last)` instead.

In https://doc.qt.io/qt-5/qset.html#fromList they added this:
>Note: Since Qt 5.14, range constructors are available for Qt's generic container classes and should be used in place of this method.

Reply:
Can we just drop these deprecated methods when 5.14 becomes minimal supported version without this intermediate changes?

Reply:
It will be a long time until we raise the minimum to Qt 5.14 and the compiler will keep emitting warnings which is annoying.
Also it the new range constructors isn't a drop-in replacement for our code since we often chain the function calls.

Reply:
>the compiler will keep emitting warnings which is annoying.

Well, it's enough.
>Also it the new range constructors isn't a drop-in replacement for our code since we often chain the function calls.

I'm not sure I understand why is it...

Anyway I really dislike such "half-named" functions like toSet. It's better to have it in corresponding namespace to be `List::toSet()`, IMO. Also I wouldn't use Q prefix in our custom code even if we just convert Qt types (like in our string utils, e.g. `String::fromDouble()` produces QString).

Reply:
>> Also it the new range constructors isn't a drop-in replacement for our code since we often chain the function calls.
>
>I'm not sure I understand why is it...

An example would be:
```c++
const QStringSet hashSet {params()["hashes"].split('|', QString::SkipEmptyParts).toSet()};  // toSet() is deprecated
```
which we will need to introduce temporary variables.

> It's better to have it in corresponding namespace to be List::toSet(), 

OK, will add namespace and rename it.

Reply:
>which we will need to introduce temporary variables.

Ok.

Reply:
>It's better to have it in corresponding namespace to be List::toSet(), IMO. Also I wouldn't use Q prefix in our custom code even if we just convert Qt types (like in our string utils, e.g. String::fromDouble() produces QString).

What about the `toQVector()` below? If following your logic it will become `Vector::toVector()` which is weird...

Reply:
>What about the toQVector() below? If following your logic it will become Vector::toVector() which is weird...

Well, seems your initial approach (toQVector, toQSet, toStdVector, etc.) is the easiest way... Let's leave it at that for now.

Reply:
>What about the toQVector() below? If following your logic it will become Vector::toVector() which is weird...

Oops... I'm confused a bit. If you mean creating QVector from std vector then it can be `Vector::fromStdVector()`.

Reply:
OK, PR updated.